### DISCUSSION TITLE
基于LLM的代码翻译系统

### DISCUSSION THREAD

**Discussion by whcadbq** [InternLM/Tutorial#2351] [project]
编程语言的快速发展以及维护遗留代码库的需求，催生了对自动化代码翻译工具的巨大需求。传统的代码翻译方法涉及大量的人工工作，需要在源语言和目标语言之间具备深厚的专业知识。随着大语言模型LLMs 的兴起，自动化代码翻译变得越来越可行。这些模型通过大量的训练数据，能够生成多种语言的代码片段，展示了在语言理解和生成任务中的卓越能力。然而，尽管取得了这些进展，语言模型在代码翻译任务中的表现仍然不稳定，尤其是在处理复杂或不常见的编程结构时其会出现许多问题，例如  a. 违法目标语言要求、b. 复制源程序代码、c. API映射错误、d. 运算符映射错误。本项目将构思合理的模型代码理解能力的评估指标，通过代码分析技术丰富输入模型的源程序信息，利用现有的微调和信息增强检索技术提高LLM的代码翻译能力。

**Comment by invisible30** [comment]
你好，是有成熟idea了吗

**Reply by whcadbq** [reply]
你好，最近看了一些文献，还在构思idea

**Comment by ranck626** [comment]
什么是代码翻译呢？java-->python，或者tensorflow-->pytorch?

**Reply by whcadbq** [reply]
目前打算做java-->python等这种类型的转换，主要是对老旧语言转化为更安全高效语言的问题的解决
